src/index.ts

```diff
@@ -785,8 +730,8 @@ export class SunsynkPowerFlowCard extends LitElement {
         let max_linewidth = (Utils.toNum(config.max_line_width) < 1 ? 1 : config.max_line_width) - 1;
         let min_linewidth = Utils.toNum(config.min_line_width) || 1;
 
-        const BatteryMaxPower = this.hass.states[config.battery.max_power] || {state: config.battery.max_power ?? ''};
-        let BattMaxPower = Utils.toNum(BatteryMaxPower.state);
+        const BatteryMaxPower = this.getEntity('max_power', {state: config.battery.max_power ?? ''});
```

Comment:
Will this work. max_power is defined under multiple config sections i.e. solar, battery etc. How do we ensure it matches on battery.max_power

Reply:
I made a change to better access the relevant objects property.